### DISCUSSION TITLE
Why is Promdump not suitable for production environments?

### DISCUSSION THREAD

**Discussion by TijmenvanGroezen** [ihcsim/promdump#15] [Q&A]
I'm looking for way to export data from a Prometheus database and found Promdump as an option. I'm only wondering what the reason behind the statement on the github page is that the application is not suitable for production environments? What are the reasons behind this statement?

**Comment by ihcsim** [comment]
I want to make sure that people don't think this is a tool for backing up and restoring Prometheus in production. In addition to the [limitations](https://github.com/ihcsim/promdump#limitations) and some of the [post-import manual steps](https://github.com/ihcsim/promdump#faq), the export is done on a live volume, where Prometheus can change the data while the export is happening.

**Reply by TijmenvanGroezen** [reply]
So could you say it is safe to backup data from a live system, but restoring the data may raise issues in production? Or could the backup also raise issues in a live system?

**Reply by ihcsim** [reply]
Backup is fine. It's read-only, so it won't mess up your Prometheus.

Restore works most of the time. On OpenShift, you might get this error on the restored Prometheus:

```
opening storage failed: get segment range: segments are not sequential
```

The [FAQ](https://github.com/ihcsim/promdump#faq) shows how to get around it.

Also, keep in mind that pre-restore, promdump will wipe the existing data on the target Prometheus. So make sure there isn't something there that you will like to retain.
